packages/@guardian/source-foundations/src/typography/data.ts

```diff
@@ -1,241 +1,153 @@
-	medium: fontSizes[9], //50px
-	large: fontSizes[10], //70px
-} as const;
-
-const headlineSizes = {
-	xxxsmall: fontSizes[3], //17px
-	xxsmall: fontSizes[4], //20px
-	xsmall: fontSizes[5], //24px
-	small: fontSizes[6], //28px
-	medium: fontSizes[7], //34px
-	large: fontSizes[8], //42px
-	xlarge: fontSizes[9], //50px
-} as const;
-
-const bodySizes = {
-	xsmall: fontSizes[1], //14px
-	small: fontSizes[2], //15px
-	medium: fontSizes[3], //17px
-} as const;
-
-const textSansSizes = {
-	xxsmall: fontSizes[0], //12px
-	xsmall: fontSizes[1], //14px
-	small: fontSizes[2], //15px
-	medium: fontSizes[3], //17px
-	large: fontSizes[4], //20px
-	xlarge: fontSizes[5], //24px
-	xxlarge: fontSizes[6], //28px
-	xxxlarge: fontSizes[7], //34px
-} as const;
-
-const fontSizeMapping = {
-	titlepiece: titlepieceSizes,
-	headline: headlineSizes,
-	body: bodySizes,
-	textSans: textSansSizes,
-} as const;
-
-const remFontSizes = fontSizes.map((fontSize) => pxToRem(fontSize));
-
-const remTitlepieceSizes = {
-	small: remFontSizes[8], //42px
-	medium: remFontSizes[9], //50px
-	large: remFontSizes[10], //70px
-} as const;
-
-const remHeadlineSizes = {
-	xxxsmall: remFontSizes[3], //17px
-	xxsmall: remFontSizes[4], //20px
-	xsmall: remFontSizes[5], //24px
-	small: remFontSizes[6], //28px
-	medium: remFontSizes[7], //34px
-	large: remFontSizes[8], //42px
-	xlarge: remFontSizes[9], //50px
-} as const;
-
-const remBodySizes = {
-	xsmall: remFontSizes[1], //14px
-	small: remFontSizes[2], //15px
-	medium: remFontSizes[3], //17px
-} as const;
-
-const remTextSansSizes = {
-	xxsmall: remFontSizes[0], //12px
-	xsmall: remFontSizes[1], //14px
-	small: remFontSizes[2], //15px
-	medium: remFontSizes[3], //17px
-	large: remFontSizes[4], //20px
-	xlarge: remFontSizes[5], //24px
-	xxlarge: remFontSizes[6], //28px
-	xxxlarge: remFontSizes[7], //34px
+/**
+ * Pixel size values for each font that we use in the design system.
+ *
+ * These fit the following px size scale:
+ * [12, 14, 15, 17, 20, 24, 28, 34, 42, 50, 70]
+ */
+export const pxTextSizes = {
+	textSans: {
+		xxsmall: 12,
+		xsmall: 14,
+		small: 15,
+		medium: 17,
+		large: 20,
+		xlarge: 24,
+		xxlarge: 28,
+		xxxlarge: 34,
+	},
+	body: {
+		xsmall: 14,
+		small: 15,
+		medium: 17,
+	},
+	headline: {
+		xxxsmall: 17,
+		xxsmall: 20,
+		xsmall: 24,
+		small: 28,
+		medium: 34,
+		large: 42,
+		xlarge: 50,
+	},
+	titlepiece: {
+		small: 42,
+		medium: 50,
+		large: 70,
+	},
 } as const;
 
-const remFontSizeMapping = {
-	titlepiece: remTitlepieceSizes,
-	headline: remHeadlineSizes,
-	body: remBodySizes,
-	textSans: remTextSansSizes,
+/**
+ * Relative font sizes, calculated from the pixel sizes above;
+ * using the pxToRem method.
+ *
+ * The resulting scale that we draw from is:
+ * [0.75, 0.875, 0.9375, 1.0625, 1.25, 1.5, 1.75, 2.125, 2.625, 3.125, 4.375]
+ *
+ *
+ * See {@link [pxToRem](../utils/px-to-rem.ts)} for more details.
+ */
+export const remTextSizes = {
+	textSans: {
+		xxsmall: pxToRem(12),
```

Comment:
does this become safer doing something like this?

```suggestion
		xxsmall: pxToRem(pxTextSizes.textSans.xxsmall),
```

Reply:
nice point, i agree it is safer — it made me think of the point I raised in chat around using concrete values instead of computing them at all. would be great to get feedback on that idea too :)

Reply:
yep 👍 subsequent PR makes sense

Reply:
changed in: [`ac5e32f` (#1500)](https://github.com/guardian/source/pull/1500/commits/ac5e32f9f6a7e3f4ee4d2eb0ac6fc55fec01ab35)